modules/prediction/prediction.cc

```diff
@@ -124,13 +122,29 @@ void Prediction::OnPerception(const PerceptionObstacles& perception_obstacles) {
       PredictorManager::instance()->prediction_obstacles();
   AdapterManager::FillPredictionHeader(Name(), &prediction_obstacles);
   AdapterManager::PublishPrediction(prediction_obstacles);
-  ADEBUG << "Published a prediction message ["
-         << prediction_obstacles.ShortDebugString() << "].";
+  for (auto const& prediction_obstacle :
+       prediction_obstacles.prediction_obstacle()) {
+    for (auto const& trajectory : prediction_obstacle.trajectory()) {
+      for (auto const& trajectory_point : trajectory.trajectory_point()) {
+        CHECK(IsValidTrajectoryPoint(trajectory_point));
+      }
+    }
+  }
 }
 
 Status Prediction::OnError(const std::string& error_msg) {
   return Status(ErrorCode::PREDICTION_ERROR, error_msg);
 }
 
+bool Prediction::IsValidTrajectoryPoint(
+    const TrajectoryPoint& trajectory_point) {
+  return (!std::isnan(trajectory_point.path_point().x())) &&
```

Comment:
check trajectory_point.has_path_point()

Reply:
This piece of code is to capture anything abnormal in real road test as I am not able to reproduce the nan issue reported in 101 test with the bags collected. Therefore if trajectory_point has no path_point, it will be captured and a core dump will be thrown. This piece of code will be removed with enough tests/verifications.